tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -105,6 +106,10 @@ type Store interface {
 
 type storeImpl struct {
     db *pgxpool.Pool
+
+    // Lock since copyFrom requires a delete first before being executed we can get in odd states if
```

Comment:
Can you please explain what kind of issue did you see? Like the sequence of operations? Does it only happen to UpsertMany conflicts?

Reply:
well @md2119 saw it.  But in generally if separate threads are updating the same rows we could get into a variety of weird states.  Since the copyFrom does not deal with on conflict it requires a manual delete first before using the copyFrom the flow is:

delete rows
insert rows with copyFrom

if we have multiple threads processing subsets of rows we can process things in a weird order like
delete - thread1
delete - thread2
copyFrom -thread2
copyFrom -thread1
in that example it really doesn't matter what order the copyFrom occurs in, since both threads have deleted, we can get into a situation where we are trying to overwrite with copyFrom and that would hit us with a unique constraint violation.  I saw this was actually happening in the bespoke Image store and wrote ROX-11050 to account for that.

Hopefully all those words clears that up a little bit.